app/models/user.rb

```diff
@@ -558,21 +560,26 @@ def update_pseud_name
     if login.downcase == login_before_last_save.downcase
       old_pseud.name = login
       old_pseud.save!
-      reindex_user_pseuds
     else
       new_pseud = pseuds.where(name: login).first
       # do nothing if they already have the matching pseud
-      return if new_pseud.present?
-      if old_pseud.present?
-        # change the old pseud to match
-        old_pseud.name = login
-        old_pseud.save!(validate: false)
-        reindex_user_pseuds
-      else
-        # shouldn't be able to get here, but just in case
-        Pseud.create!(name: login, user_id: id)
+      if new_pseud.blank?
+        if old_pseud.present?
+          # change the old pseud to match
+          old_pseud.name = login
+          old_pseud.save!(validate: false)
+        else
+          # shouldn't be able to get here, but just in case
+          Pseud.create!(name: login, user_id: id)
+        end
       end
     end
+
+    # Works, series, and pseuds are indexed with the user's byline,
+    # which has the old username, so they all need to be reindexed.
```

Comment:
I was going to ask about bookmarks, but honestly, the issue just says "works" so we can fix bookmarks in 0.9.212 >.>

Reply:
Once more with feelings (and bookmarks)! #3300.